internal/node/node.go

```diff
@@ -218,52 +223,68 @@ func (n *Node) generateKeypair() error {
 
 func (n *Node) init() error {
 	var err error
+	var _err error
 
 	n.initOnce.Do(func() {
-		err = n.loadNodeConfig()
-		if err != nil {
-			n.log.Error("Failed to load node configuration file", slog.Any("err", err), slog.String("config_path", n.nodeOpts.ConfigFilepath))
+		_err = n.loadNodeConfig()
+		if _err != nil {
+			n.log.Error("Failed to load node configuration file", slog.Any("err", _err), slog.String("config_path", n.nodeOpts.ConfigFilepath))
+			err = errors.Join(err, _err)
 		} else {
 			n.log.Info("Loaded node configuration", slog.String("config_path", n.nodeOpts.ConfigFilepath))
 		}
 
-		n.telemetry, err = observability.NewTelemetry(n.ctx, n.log, n.config, n.publicKey)
-		if err != nil {
-			n.log.Error("Failed to initialize telemetry", slog.Any("err", err))
+		n.telemetry, _err = observability.NewTelemetry(n.ctx, n.log, n.config, n.publicKey)
+		if _err != nil {
+			n.log.Error("Failed to initialize telemetry", slog.Any("err", _err))
+			err = errors.Join(err, _err)
 		} else {
 			n.log.Info("Telemetry status", slog.Bool("metrics", n.config.OtelMetrics), slog.Bool("traces", n.config.OtelTraces))
 		}
 
+		// start public NATS server
+		_err = n.startPublicNATS()
+		if _err != nil {
+			n.log.Error("Failed to start public NATS server", slog.Any("err", _err))
+			err = errors.Join(err, fmt.Errorf("failed to start public NATS server: %s", _err))
+		} else if n.natspub != nil {
+			n.log.Info("Public NATS server started", slog.String("client_url", n.natspub.ClientURL()))
+		}
+
 		// setup NATS connection
-		n.nc, err = models.GenerateConnectionFromOpts(n.opts)
-		if err != nil {
-			n.log.Error("Failed to connect to NATS server", slog.Any("err", err))
-			err = fmt.Errorf("failed to connect to NATS server: %s", err)
+		n.nc, _err = models.GenerateConnectionFromOpts(n.opts)
+		if _err != nil {
+			n.log.Error("Failed to connect to NATS server", slog.Any("err", _err))
+			err = errors.Join(err, fmt.Errorf("failed to connect to NATS server: %s", _err))
 		} else {
 			n.log.Info("Established node NATS connection", slog.String("servers", n.opts.Servers))
 		}
 
-		// init internal NATS server
-		err = n.startInternalNATS()
-		if err != nil {
-			n.log.Error("Failed to start internal NATS server", slog.Any("err", err))
-			err = fmt.Errorf("failed to start internal NATS server: %s", err)
+		// start internal NATS server
+		_err = n.startInternalNATS()
```

Comment:
this looks a little awkward. The underscore prefix typically indicates that the variable isn't being used, but this variable is used in multiple places.